compiler/ir/backend.common/src/org/jetbrains/kotlin/backend/common/lower/loops/Utils.kt

```diff
@@ -23,9 +23,14 @@ internal fun IrExpression.castIfNecessary(targetType: IrType, numberCastFunction
     return if (type == targetType) {
         this
     } else {
-        val function = type.getClass()!!.functions.first { it.name == numberCastFunctionName }
-        IrCallImpl(startOffset, endOffset, function.returnType, function.symbol)
-            .apply { dispatchReceiver = this@castIfNecessary }
+        // This expression's type could be Nothing from an exception throw, in which case the number cast function will not exist.
+        val castFun = type.getClass()!!.functions.firstOrNull { it.name == numberCastFunctionName }
+        return if (castFun != null) {
```

Comment:
@punzki Is it possible to switch null check to Nothing type check that mentioned in comment?

Reply:
Good idea. Done!